### PR TITLE
[RFC] Correctly parse Doctrine SQL array

### PR DESCRIPTION
**Pull request by aschempp**
Since Contao 4.3 we can set an array for the SQL definition in DCAs. However, the default value and other configs were not correctly detected/merged.

### CODE REVIEW

**Comment by qzminski** [comment]
It should also support arrays here:

https://github.com/aschempp/contao-core-bundle/blob/a0c675b4741742100f112bfa714395289974ec8f/src/Resources/contao/library/Contao/Widget.php#L1525

**Comment by aschempp** [comment]
Fixed in 5f9207f69536ee0750f7b7ca709726b244b5e02f

File: src/Resources/contao/library/Contao/Widget.php [hunk 1]
```diff
@@ -1484,20 +1525,12 @@ public function getEmptyStringOrNull()
...
-		// Strip the field type definition
-		list(, $def) = explode(' ', $sql, 2);
-
-		if (strpos($def, 'NULL') !== false && strpos($def, 'NOT NULL') === false)
-		{
-			return null;
-		}
-
-		return '';
+		return static::getEmptyValueByFieldType($sql) === null ? null : '';
```
Comment (leofeyer) [inline src/Resources/contao/library/Contao/Widget.php]: This change is wrong. The method comment says:

> Return either an empty string or null based on the SQL string

Therefore we cannot use `getEmptyValueByFieldType()` here!

**Comment by aschempp** [comment]
getEmptyValueByFieldType return null, empty string or zero, and zero would be converted to empty string?

**Comment by leofeyer** [comment]
Yes, there is an `getEmptyStringOrNull()` method using the `getEmptyStringOrNullByFieldType()` method internally. And even though I do not remember why we need this method, we must not change the API in a bugfix release.

**Comment by aschempp** [comment]
Why do you think this is an API change? The method still exists, and it returns the same thing as before?

**Comment by leofeyer** [comment]
:open_mouth: My bad.

**Pull request merged**